ui/src/AzureUserInput.ts

```diff
@@ -0,0 +1,104 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Memento, QuickPickItem, QuickPickOptions } from 'vscode';
+import * as vscode from 'vscode';
+import { IAzureQuickPickItem, IAzureQuickPickOptions, IAzureUserInput } from '../index';
+import { DialogResponses } from './DialogResponses';
+import { UserCancelledError } from './errors';
+import { localize } from './localize';
+import { randomUtils } from './utils/randomUtils';
+
+export class AzureUserInput implements IAzureUserInput {
+    private readonly _persistence: Memento;
+
+    public constructor(persistence: Memento) {
+        this._persistence = persistence;
+    }
+
+    public async showQuickPick<T extends QuickPickItem>(items: T[] | Thenable<T[]>, options: QuickPickOptions): Promise<T> {
+        if (options.ignoreFocusOut === undefined) {
+            options.ignoreFocusOut = true;
+        }
+
+        let persistenceKey: string | undefined;
+        const hashData: string | undefined = (<IAzureQuickPickOptions>options).id || options.placeHolder;
```

Comment:
We should add something from the wizard itself into this hash, otherwise different wizards asking for resource groups will all persistent the same choice.

Reply:
Two things:
1. IMO that's a good thing. If I select a resource group for one wizard, I want that to also be 'recently used' in another wizard
1. GlobalState is on a per-extension basis, so this already only works on a per-extension basis

Reply:
2. is good, but I'm not convinced about 1.  Sometimes it will be good, sometimes not. If you don't want it for a particular one you can change the behavior by manually giving an id, right? So I'm willing to wait and see how it works out.

Reply:
Yes you can change the behavior by specifying an id